### PR TITLE
polished sonoff th10 documentation

### PR DESCRIPTION
**Pull request by reogaro**
I reworked and corrected the Sonoff TH10 documentation, and added some notes on its almost-identical bigger brother, the TH16.

Please be nice, this is my first pull request.

## Type of changes

- [ ] New device
- [x] Update existing device
- [ ] Removing a device
- [x] General cleanup
- [ ] Other

## Checklist:

- [x] There are no passwords or secrets references in any examples. 
      The only exceptions are `!secret wifi_ssid` and `!secret wifi_password`.
- [x] The `wifi` or `ethernet` block has no static / manual ip address specified.
- [x] The first configuration provided should be **hardware definitions only**.
      A more involved example can be provided in a separate configuration block.

### CODE REVIEW

File: src/docs/devices/Sonoff-TH10/index.md [hunk 1]
```diff
@@ -1,76 +1,93 @@
...
+  password: !secret wifi_password
+
+# One-Wire for 2.5mm jack sensors
+# one_wire:
+# - platform: gpio
+# pin: GPIO14
+
+sensor:
+  - platform: wifi_signal
+    name: "${friendly_name} WiFi Signal"
```
Comment (jesserockz) [inline src/docs/devices/Sonoff-TH10/index.md]: Entities should not be prefixed with friendly name substitution as it is added automatically by using `esphome` -> `friendly_name`

```suggestion
    name: WiFi Signal
```

File: src/docs/devices/Sonoff-TH10/index.md [hunk 2]
```diff
@@ -1,76 +1,93 @@
...
-    update_interval: 60s
 switch:
   - platform: restart
-    name: "Sonoffth10_1 Restart"
+    name: "${friendly_name} Restart"
 
 output:
   - platform: esp8266_pwm
-    id: TH10_1_blue_led
+    id: TH1x_1_blue_led
```
Comment (jesserockz) [inline src/docs/devices/Sonoff-TH10/index.md]: IDs can be simpler
```suggestion
    id: blue_led
```

File: src/docs/devices/Sonoff-TH10/index.md [hunk 3]
```diff
@@ -1,76 +1,93 @@
...
+  friendly_name: "Sonoff TH1X"
+
 # Basic Config
 esphome:
-  name: sonoffth10_1
+  name: "${name}"
+  friendly_name: "${friendly_name}"
   platform: ESP8266
   board: esp01_1m
   board_flash_mode: dout
```
Comment (jesserockz) [inline src/docs/devices/Sonoff-TH10/index.md]:
```suggestion

esp8266:
  board: esp01_1m
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
A few suggestions below and will need more manual changes to align ids too

**Comment by reogaro** [comment]
I did the requested changes - is this up to spec?

**Pull request merged**